toxcore/net_crypto.c

```diff
@@ -40,23 +43,32 @@ typedef enum Crypto_Conn_State {
     CRYPTO_CONN_NO_CONNECTION,       /* the connection is allocated, but not yet used */
     CRYPTO_CONN_COOKIE_REQUESTING,   /* we are sending cookie request packets */
     CRYPTO_CONN_HANDSHAKE_SENT,      /* we are sending handshake packets */
-    CRYPTO_CONN_NOT_CONFIRMED,       /* we are sending handshake packets.
+    CRYPTO_CONN_NOT_CONFIRMED,       /* we are sending handshake packets. // AKE: accepted
                                       * we have received one from the other, but no data */
-    CRYPTO_CONN_ESTABLISHED,         /* the connection is established */
+    CRYPTO_CONN_ESTABLISHED,         /* the connection is established */ // AKE: confirmed
 } Crypto_Conn_State;
 
 typedef struct Crypto_Connection {
     uint8_t public_key[CRYPTO_PUBLIC_KEY_SIZE]; /* The real public key of the peer. */
-    uint8_t recv_nonce[CRYPTO_NONCE_SIZE]; /* Nonce of received packets. */
-    uint8_t sent_nonce[CRYPTO_NONCE_SIZE]; /* Nonce of sent packets. */
-    uint8_t sessionpublic_key[CRYPTO_PUBLIC_KEY_SIZE]; /* Our public key for this session. */
-    uint8_t sessionsecret_key[CRYPTO_SECRET_KEY_SIZE]; /* Our private key for this session. */
-    uint8_t peersessionpublic_key[CRYPTO_PUBLIC_KEY_SIZE]; /* The public key of the peer. */
+    //AKE NEW: nonces not needed for Noise => handshake object
```

Comment:
These commented out things need to be cleaned up.

How much work would it be to make the new protocol optional (either by compile time flag or a Tox_Options field)?